src/main/java/io/github/seggan/kingoftheziggurat/Ziggurat.java

```diff
@@ -61,10 +63,26 @@ private void tick() {
             bot.strength++;
             bot.direction = MoveDirection.NONE;
         }
+        // group bots by position
+        HashMap<Point, ArrayList<Bot>> botsByPos = new HashMap<>();
```

Comment:
Use interface types please
```suggestion
        Map<Point, ArrayList<Bot>> botsByPos = new HashMap<>();
```